src/detail/header.hpp

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright (c) 2024 Christian Mazakas
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CPPAlliance/http_proto
+//
+
+#ifndef BOOST_HTTP_PROTO_SRC_DETAIL_HEADER_HPP
+#define BOOST_HTTP_PROTO_SRC_DETAIL_HEADER_HPP
+
+#include <boost/http_proto/detail/except.hpp>
+#include <boost/http_proto/message_base.hpp>
+#include <boost/http_proto/detail/header.hpp>
+#include <boost/core/span.hpp>
+#include <cstddef>
+
+namespace boost {
+namespace http_proto {
+namespace detail {
+
+struct prefix_op
+{
+    message_base& mb_;
+    span<char> prefix_;
+    char* buf_ = nullptr;
+    std::size_t n_ = 0;
+
+    prefix_op(
+        message_base& mb,
+        std::size_t n)
```

Comment:
would noexcept help?

Reply:
We shouldn't apply `noexcept` here because this function allocates and it also enforces a length check via `detail::throw_length_error()`.